hashicorp/terraform-provider-tfe
# Data retention policy

## Description

Add data retention policy attributes to organization and workspace resources

_Remember to:_


## Testing plan

In addition to automated tests, manually plan/apply config that looks something like this:

```
resource "tfe_organization" "test-organization" {
  name  = "my-org-name"
  email = "admin@company.com"

  data_retention_policy {
    delete_older_than_n_days = 12
  }
}

resource "tfe_workspace" "test-workspace" {
  name = "drp-workspace"
  organization = tfe_organization.test-organization.name

  data_retention_policy {
    delete_older_than_n_days = 8
  }
}
```

* Ensure that creating/updating/deleting data retention policies works fine
* Ensure that workspaces/orgs with data retention policies are imported correctly
* Ensure the provider works properly when run against TFE/TFC that doesn't have data retention policy support

## External links

- [go-tfe documentation](https://pkg.go.dev/github.com/hashicorp/go-tfe#DataRetentionPolicy)

## Output from acceptance tests

_Please run applicable acceptance tests locally and include the output here. See [testing.md](https://github.com/hashicorp/terraform-provider-tfe/blob/main/docs/testing.md) to learn how to run acceptance tests._

_If you are an external contributor, your contribution(s) will first be reviewed before running them against the project's CI pipeline._

```
$ TESTARGS="-run TestAccTFEWorkspace" make testacc

...
```
